test/Utilities/Base32.spec.ts

```diff
@@ -0,0 +1,188 @@
+                'PV7H7AEBQKBYJBMGQ6EITCULRSGY5D4QSGJJHFEV' +
+                'S2LZRGM2TOOJ3HU7UCQ2FI5EUWTKPKFJVKV2ZLNO' +
+                'V6YLDMVTWS23NN5YXG5LXPF5X274BQOCYPCMLRWH' +
+                'ZDE4VS6MZXHM7UGR2LJ5JVOW27MNTWW33TO55X7A' +
+                '4HROHZHF43T6R2PK5PWO33XP6DY7F47U6X3PP6HZ' +
+                '7L57Z7P674AACAQD';
+            expect(encoded).to.equal(expected);
+        });
+
+        it('throws if input size is not a multiple of block size', () => {
+            // Arrange:
+            for (let i = 2; 10 > i; i += 2) {
+                const input = new Uint8Array(i);
+
+                // Act + Assert:
+                expect(() => {
+                    Base32.Base32Encode(input);
+                }, `input at ${i}`).to.throw('decoded size must be multiple of 5');
+            }
+        });
+    });
+
+    describe('decode', () => {
+        it('can convert empty input', () => {
+            // Act:
+            const decoded = Base32.Base32Decode('');
+
+            // Assert:
+            expect(Converter.uint8ToHex(decoded)).to.equal('');
+        });
+
+        it('can convert test vectors', () => {
+            // Arrange:
+            for (const sample of Test_Vectors) {
+                // Act:
+                const decoded = Base32.Base32Decode(sample.encoded);
+
+                // Assert:
+                expect(Converter.uint8ToHex(decoded), `input ${sample.encoded}`).to.equal(sample.decoded);
+            }
+        });
+
+        it('accepts all valid characters', () => {
+            // Act:
+            const decoded = Base32.Base32Decode('ABCDEFGHIJKLMNOPQRSTUVWXYZ234567');
+
+            // Assert:
+            expect(Converter.uint8ToHex(decoded)).to.equal('00443214C74254B635CF84653A56D7C675BE77DF');
+        });
+
+        it('throws if input size is not a multiple of block size', () => {
+            // Arrange:
+            for (let i = 1; 8 > i; ++i) {
+                const input = 'A'.repeat(i);
+
+                // Act + Assert:
+                expect(() => {
+                    Base32.Base32Decode(input);
+                }, `input at ${i}`).to.throw('encoded size must be multiple of 8');
+            }
+        });
+
+        it('throws if input contains an invalid character', () => {
+            // Arrange:
+            const illegalInputs = [
+                'NC5J5DI2URIC4H3T3IMXQS21PWQWZIPEV6EV7LAS', // contains char '1'
+                'NBGCMBPFWNTLXFF4GB2V5SPV!V2OQD6JFA6SBYUD', // contains char '!'
+                'NDL3BGQUX2T44BQOOHAPVGWJWQRG3YLHAE)6CCZ5', // contains char ')'
+            ];
+
+            // Act + Assert:
+            for (const input of illegalInputs) {
+                expect(() => {
+                    Base32.Base32Decode(input);
+                }, `input ${input}`).to.throw('illegal base32 character');
+            }
+        });
+    });
+
+    describe('roundtrip', () => {
+        it('decode -> encode', () => {
+            // Arrange: inputs
```

Comment:
maybe we can remove the `inputs` because in most cases we leave it empty after `Arrange:`